### PR TITLE
[Dominion] Country Page Hero

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

Dominion profile page along with project guidelines.

NOTE: THIS PR RESOLVES A MERGE CONFLICT

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots

screen shot 2019-03-04 at 4 02 54 pm

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: dominion/templates/profile/profile_detail.html [hunk 1]
```diff
@@ -1,44 +1,12 @@
...
-            <span>Sectional Title Land</span></a></li>
-        </ul>
-      </div>
-    </nav>
-  </div>
-{% endblock %}
+{% extends 'profile/profile_detail.html' %} {% load humanize lookup staticfiles hashtospace %} {% load jsonify %}
+
+{% block header %}
+    {% include 'blocks/header.html' %}
```
Comment (kilemensi) [inline dominion/templates/profile/profile_detail.html]: Small fix: header needs to change icon when viewed in a country profile.

screenshot from 2019-03-05 14-17-39

File: dominion/templates/profile/profile_detail.html [hunk 2]
```diff
@@ -1,44 +1,12 @@
...
+{% endblock header %}
+
+{% block content_container %}
+  <div id="dominionCountryHero"></div>
+{% endblock content_container %}
```
Comment (kilemensi) [inline dominion/templates/profile/profile_detail.html]: Small fix: Wazimap already defines `{% block content %}` in `profile_detail.html`. Why don't we reuse it?

File: dominion/static/js/profile_map_mapit.js [hunk 3]
```diff
@@ -75,7 +74,7 @@ var ProfileMaps = function() {
 
     this.addImagery = function() {
         // add imagery
-        L.tileLayer('//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+        L.tileLayer('//{s}.tiles.wmflabs.org/bw-mapnik/{z}/{x}/{y}.png', {
```
Comment (kilemensi) [inline dominion/static/js/profile_map_mapit.js:91]: QQ: Is `tiles.wmflabs.org/bw-mapnik/` a free service, dependable, etc?

File: dominion/templates/profile/profile_detail.html [hunk 4]
```diff
@@ -56,6 +24,22 @@
       <script src="{% static 'js/profile_map.js' %}"></script>
       <script src="{% static 'js/maps_mapit.js' %}"></script>
       <script src="{% static 'js/profile_map_mapit.js' %}"></script>
+      <script src="{% static 'js/map_mapit.js' %}"></script>
```
Comment (kilemensi) [inline dominion/templates/profile/profile_detail.html]: Why both `'js/map_mapit.js'` and `'js/maps_mapit.js'`?

File: dominion/dominion_ui/src/components/Hero/CountryHero.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    top: 'unset !important',
+    [theme.breakpoints.up('sm')]: {
+      position: 'absolute',
+      right: '150px'
+    },
+    [theme.breakpoints.down('sm')]: {
+      position: 'relative !important',
+      width: '100% !important',
+      height: '250px !important'
+    }
```
Comment (kilemensi) [inline dominion/dominion_ui/src/components/Hero/CountryHero.js:34]: The above two `[theme.breakpoints.up('sm')]` and `[theme.breakpoints.down('sm')]` clashes for `sm` i.e. break points work from that point up or down _inclusively_.

**Review by kilemensi — CHANGES_REQUESTED** [review]
Almost there @karimkawambwa.

Big issues are:
 1. For this design (and most of our designs), consider anything from `sm` and below as mobile. These two images one `sm` and the other `xs` should look the same.
screenshot_2019-03-05 dominion profile south africa
screenshot_2019-03-05 dominion profile south africa 1
 2. Scale `md` to look like `lg` but _smaller_. Otherwise components overlap.
screenshot_2019-03-05 dominion profile south africa 2

File: dominion/templates/profile/profile_detail.html [hunk 6]
```diff
@@ -56,6 +24,22 @@
       <script src="{% static 'js/profile_map.js' %}"></script>
```
Comment (KhadijaMahanga) [inline dominion/templates/profile/profile_detail.html]: let us remove `body_javascript` block  on profile html, as we have moved to mapit

Comment (karimkawambwa) [reply dominion/templates/profile/profile_detail.html]: not sure what you mean, do you mean not use `profile_map.js`?

File: dominion/templates/profile/profile_detail.html [hunk 7]
```diff
@@ -56,6 +24,22 @@
...
+
+            var COUNTRY = {{ geography.this|jsonify|safe }};
+            var centre = {
+                'country-KE': [-0.0236, 37.9062],
+                'country-ZA': [-30.5595, 22.9375],
+                'country-TZ': [-6.3690, 34.8888],
+                'country-NG': [9.0820, 8.6753],
+            }
+
+            maps.drawMapForHomepage('{{ geo_data.first_child_level }}', '{{ HURUMAP.default_geo_version }}', centre['{{ geography.this.full_geoid }}'], zoom);
```
Comment (KhadijaMahanga) [inline dominion/templates/profile/profile_detail.html]: `profile_detail ` from hurumap also defines `a ProfileMaps` object. I think we should move to it's own template, once we compile a complete country page

File: dominion/templates/profile/profile_detail.html (hunk 2 above)
Comment (karimkawambwa) [reply dominion/templates/profile/profile_detail.html]: content has unwanted wrappers

File: dominion/static/js/profile_map_mapit.js (hunk 3 above)
Comment (karimkawambwa) [reply dominion/static/js/profile_map_mapit.js:91]: https://wiki.openstreetmap.org/wiki/Tile_servers @kilemensi

File: dominion/templates/profile/profile_detail.html (hunk 1 above)
Comment (karimkawambwa) [reply dominion/templates/profile/profile_detail.html]: done

**Comment by karimkawambwa** [comment]
@kilemensi @KhadijaMahanga 👀

File: dominion/dominion_ui/public/index.html [hunk 8]
```diff
@@ -59,6 +59,14 @@
...
     <script type="text/javascript">
         var MAPIT = { url: "https://mapit.hurumap.org" };
+        var dominion_countries = {
+            'south-africa': {
+                name: 'South Africa'
+            },
+            'kenya': {
+                name: 'Kenya'
+            }
+        }
```
Comment (KhadijaMahanga) [inline dominion/dominion_ui/public/index.html:69]: These are defined on dominion template (`settings_js.html`), they shouldn't be defined again here.

 Or are they here to run front-end as a stand alone using `yarn start`

Comment (karimkawambwa) [reply dominion/dominion_ui/public/index.html:69]: yeah for `yarn start`

File: dominion/dominion_ui/public/index.html [hunk 9]
```diff
@@ -59,14 +59,23 @@
...
     </script>
   </head>
 
-  <body>
+  <body style="background-image: url(/static/img/background.png);">
```
Comment (KhadijaMahanga) [inline dominion/dominion_ui/public/index.html]: @karimkawambwa I am sure this works running dominion_ui as a standalone. If you run dominion hurumap-app, we don't have a background :open_mouth: 
Either we add this css for dominion (django app) templates  or have it set on the components
screenshot from 2019-03-07 13-17-52

**Review by KhadijaMahanga — CHANGES_REQUESTED** [review]

File: dominion/dominion_ui/src/components/Hero/HomeHero.js [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+            Dominion makes data available to help add context and authority to
+            public discourse and policy-making on vital issues of land
+            ownership.
+          </HeroDescription>
+
+          <PortalChooser
+            activator={({ handleToggle }) => (
+              <HeroButton onClick={handleToggle}>Select a Country</HeroButton>
+            )}
+          />
```
Comment (KhadijaMahanga) [inline dominion/dominion_ui/src/components/Hero/HomeHero.js]: The modal portal chooser, from `select a country` does not change the arrow down/up on the nav. Is `handleToggle` passed on HeroButton initialised on HomeHero?

File: dominion/dominion_ui/public/index.html (hunk 9 above)
Comment (karimkawambwa) [reply dominion/dominion_ui/public/index.html]: @KhadijaMahanga 

Pushed the changes already.
I forgot the setup @kilemensi had does not import css from react app.

File: dominion/dominion_ui/src/components/Hero/HomeHero.js (hunk 10 above)
Comment (karimkawambwa) [reply dominion/dominion_ui/src/components/Hero/HomeHero.js]: I'll fix that, i might've missed that part.

File: dominion/dominion_ui/public/index.html (hunk 9 above)
Comment (kilemensi) [reply dominion/dominion_ui/public/index.html]: Mhh... we don't _css_ in react @karimkawambwa, we _css in js_. This feels like the background should be applied to the header component _in react_ to me (header = nav + hero).

**Review by kilemensi — CHANGES_REQUESTED** [review]
@karimkawambwa:

 1. The Home page hero breaks in large screens,
screenshot from 2019-03-07 14-58-31

 2. *Find a place* should show the search
screenshot from 2019-03-07 15-02-35

 3. Clicking `or view Country` should take you to the country profile page
screenshot from 2019-03-07 15-00-27

-------------------------------------------------

Also, if it's possible, the zoom buttons should appear at the bottom and not the top.
screenshot from 2019-03-07 15-01-23

**Comment by KhadijaMahanga** [comment]
@karimkawambwa  here are minor things I have noticed

1. Select Country on home hero still doesn't change the `countries` up/down icon on nav

2. Let's add opacity to `HeroDescription` text

screenshot from 2019-03-08 12-07-56

3. PortalChoose closeSpan in transformed to upperCase. Let's rename "Use your location" to "Use your current location"

screenshot from 2019-03-08 12-08-04

**Comment by karimkawambwa** [comment]
@KhadijaMahanga @kilemensi 👀

**Comment by KhadijaMahanga** [comment]
Screenshot from 2019-03-09 08-52-52
@karimkawambwa The search icon on nav is changing to a close icon when search modal is open. It's a new change, did I miss a memo on the design :smile_cat:

**Review by KhadijaMahanga — APPROVED** [review]

**Pull request merged**